src/core/segment_buffers/implementations/text/html/text_track_cues_store.ts

```diff
@@ -72,6 +77,17 @@ export default class TextTrackCuesStore {
             ret.push(cues[j].element);
           }
         }
+        // first or last IHTMLCue in a group can have a slighlty different start
+        // or end time than the start or end time of the ICuesGroup due to parsing
+        // approximation.
+        // Add a tolerance of 1ms to fix this issue
+        if (ret.length === 0 && cues.length) {
+          if (areNearlyEqual(time, cues[0].start, DELTA_CUES_GROUP)) {
+            ret.push(cues[0].element);
+          } else if (areNearlyEqual(time, cues[cues.length - 1].end, DELTA_CUES_GROUP)) {
+            ret.push(cues[cues.length - 1].element);
+          }
+        }
```

Comment:
Nice! Makes me wonder, the old code strangely seemed to potentially push multiple cues from a single segment if they all were parts of the same timecode.

I don't remember if that's a real case (it may be, considering that `get` returns an array of HTMLElement), but in that case, maybe we should also handle the possibility that multiple cues at the start / at the end may be displayed?

Reply:
Yes it currently push only one cues because I thought that it would be very unfortunate to have multiples cues
in this situation.

Changing it to that would be a solution, what do you think ?

```js
        if (ret.length === 0 && cues.length) {
          for (let j = 0; j < cues.length; j++) {
            if (areNearlyEqual(time, cues[j].start, DELTA_CUES_GROUP)
            || areNearlyEqual(time, cues[j].end, DELTA_CUES_GROUP)
            ) {
              ret.push(cues[j].element);
          }
        }
```

Reply:
> Yes it currently push only one cues because I thought that it would be very unfortunate to have multiples cues  in this situation.

Maybe for when there's multiple locutors' subtitles at the same timecode? I don't remember at all the rules of that structure :p

> Changing it to that would be a solution, what do you think ?

Yes that seems nice. Careful to also check `cues.length > 0` (so, explicitely) in the if, I think we have a linting rule enforcing that.

Reply:
The lint error was a missing "}"